From 29256c68152834bfb121f66f101ccbdee5d75d04 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 31 Oct 2016 12:18:41 +0000
Subject: [PATCH 17/41] Conda (macOS): Use bundleIdentifier to detect running
 from bundle

.. instead of whether or not Info.plist exists as bundleIdentifier
is more reliable (esp. when debugging).

In conda both running rstudio executable directly or from an App
bundle are supported.
---
 src/cpp/desktop-mac/AppDelegate.mm | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cpp/desktop-mac/AppDelegate.mm b/src/cpp/desktop-mac/AppDelegate.mm
index 1f38a961af..36291407b1 100644
--- a/src/cpp/desktop-mac/AppDelegate.mm
+++ b/src/cpp/desktop-mac/AppDelegate.mm
@@ -293,6 +293,7 @@ bool prepareEnvironment(Options& options)
 {
    // check for open file request (either apple event or command line)
    NSString* openFile = verifyAndNormalizeFilename(openFile_);
+   NSString* bundleId = [[NSRunningApplication currentApplication] bundleIdentifier];
    if (!openFile)
    {
       openFile = openFileCommandLineArgument();
@@ -355,7 +356,7 @@ bool prepareEnvironment(Options& options)
       scriptsPath = installPath.complete("bin");
       
       // check for running in a bundle
-      if (installPath.complete("Info.plist").exists())
+      if (bundleId != nil && !sessionPath.exists())
       {
          sessionPath_ = installPath.complete("MacOS/rsession");
          scriptsPath = installPath.complete("MacOS");
-- 
2.20.1

